serve/tests/unittest/test_engine_with_samplers.py

```diff
@@ -234,5 +233,7 @@ def _test_stop(
     _test_ignore_eos(model_artifact_path, use_staging_engine=False)
     _test_stop(model_artifact_path, use_staging_engine=False)
     _test_stop(model_artifact_path, use_staging_engine=True)
-    _test_max_context_length(model_artifact_path, use_staging_engine=True)
-    _test_max_context_length(model_artifact_path, use_staging_engine=False)
+    # These tests are broken since we are now imposing no length limit
+    # if max_tokens = None. The tests do not finish in a reasonable time.
+    # _test_max_context_length(model_artifact_path, use_staging_engine=True)
+    # _test_max_context_length(model_artifact_path, use_staging_engine=False)
```

Comment:
As of https://github.com/octoml/mlc-llm/pull/105, if `max_token` is None, we are now imposing no limit on the number of generated tokens. So we let generation continue even if `max_context_len` is reached. 

This change in the engine behavior was not intended in #105, but after discussion with @sunggg we agreed that this is the behavior we want going forward. The tests above don't set `max_tokens`, so they potentially run forever. So I disabled them. 

However, this made me wonder if this "generation without length limit" is a safe thing to do - we might end up with a non-terminating request. Do we need to worry about such case in practice, or are we sure that we can rely on stop tokens for termination? @sunggg @elvin-n

Reply:
Yep, I also have the similar concern: https://github.com/octoml/mlc-llm/pull/122/files/63a3f19ff6d9e593d0394d5850b8a1db4b89de68#r1430465659

I would like to add something related to the engine capability, but the sliding window is bit tricky since it basically allows no limit.